packages/utilities/src/bn.ts

```diff
@@ -0,0 +1,301 @@
+import BigNumber from 'bignumber.js'
+
+const WEI_DECIMALS = 18
+
+export enum BN_ROUNDING {
+  DEFAULT = 3,
+  UP = 0,
+  DOWN = 1,
+  CEIL = 2,
+  FLOOR = 3,
+  HALF_UP = 4,
+  HALF_DOWN = 5,
+  HALF_EVEN = 6,
+  HALF_CEIL = 7,
+  HALF_FLOOR = 8,
+}
+
+export interface BnCfg {
+  decimals: number
+  rounding?: BN_ROUNDING
+  noGroupSeparator?: boolean
+}
+
+export type BnFormatCfg = BigNumber.Format & BnCfg
+export type BnLike = string | number | BigNumber | BN
+
+BigNumber.config({
+  DECIMAL_PLACES: 0,
+  ROUNDING_MODE: BN_ROUNDING.DEFAULT,
+  FORMAT: {
+    decimalSeparator: '.',
+    groupSeparator: ',',
+    groupSize: 3,
+  },
+})
+
+export class BN {
+  #bn: BigNumber
+  #cfg: BnCfg
+
+  static ROUNDING = BN_ROUNDING
+  static MAX_UINT256 = BN.#instance(1).pow(256).minus(1)
+
+  protected constructor(bigLike: BnLike, cfg: BnCfg) {
+    if (!cfg.decimals || cfg.decimals < 0) {
+      throw new Error(
+        `Decimals must be greater than 0, ${JSON.stringify({
+          number: bigLike,
+          cfg,
+        })}`,
+      )
+    }
+
+    const finalCfg = {
+      decimals: cfg.decimals,
+      rounding: cfg.rounding || BN_ROUNDING.DEFAULT,
+      noGroupSeparator: cfg.noGroupSeparator || true,
+    } as BnCfg
+
+    this.#bn = BN.#instance(bigLike, finalCfg)
+    this.#cfg = finalCfg
+  }
+
+  static fromFraction(
```

Comment:
```ts
/**
 * @description @link{BN.fromFraction} accepts uint number value
 * @example
 * ```ts
 * const oneEth = BN.fromFraction('1000000000000000000', 18)
 * ```
 */
 ```
 
 add equivalent for `fromRaw`